actions/summarize_pr.py

```diff
@@ -82,11 +82,27 @@ def label_fixed_issues(event, pr_summary):
         return None
 
     comment = generate_issue_comment(data["url"], pr_summary, pr_credit, data.get("title") or "")
-
-    for issue in data["closingIssuesReferences"]["nodes"]:
-        number = issue["number"]
-        event.post(f"{GITHUB_API_URL}/repos/{event.repository}/issues/{number}/labels", json={"labels": ["fixed"]})
-        event.post(f"{GITHUB_API_URL}/repos/{event.repository}/issues/{number}/comments", json={"body": comment})
+    closing_issues = data["closingIssuesReferences"]["nodes"]
+    if not closing_issues:
+        return pr_credit
+
+    # Batch get all issue node IDs in single query
+    issue_numbers = [issue["number"] for issue in closing_issues]
+    issue_node_ids = event.get_multiple_issue_node_ids(issue_numbers)
+
+    # Get label IDs once
+    label_ids = event.get_label_ids(["fixed"])
+    if not label_ids:
+        print("Warning: 'fixed' label not found")
+        return pr_credit
+
+    # Batch label and comment all issues using GraphQL mutations
+    for issue_num, issue_node_id in issue_node_ids.items():
```

Comment:
💭 **SUGGESTION**: `issue_num` is not used in this loop, which can be slightly confusing. Iterating directly over the values makes the intent clearer.

**Suggested change:**
```suggestion
    for issue_node_id in issue_node_ids.values():
```